### ISSUE TITLE
Softmax with dim=4 got different result with Keras softmax

### ISSUE THREAD

**Issue by dding3** [intel-analytics/analytics-zoo#961]

**Comment by hkvision** [comment]
Torch softmax for >=3D is not the same as Keras softmax in terms of dimension. For 3D, I use transpose to make them the same.
From keras:
```
softmax: Softmax applied across inputs last dimension. Expects shape either (nb_samples, nb_timesteps, nb_dims) or (nb_samples, nb_dims)
```
Does softmax takes 4D input?

**Comment by dding3** [comment]
Yes. No worries, I have fixed it :)

**Comment by hkvision** [comment]
Thanks so much!
